### PR TITLE
Feature/upload challenge

### PR DESCRIPTION
**Pull request by lorenzschreyer**
implemented fileUpload and githubApi integration

### CODE REVIEW

**Review by KRIYO144 — APPROVED** [review]

File: backend/codingchallenge/api/urls.py [hunk 1]
```diff
@@ -22,4 +23,7 @@
     path('application/submitChallenge', applications.SubmitApplicationView.as_view()),
     path('application/startChallenge', applications.StartChallengeView.as_view()),
     path('application/challenges', challenges.ApplicationChallengesView.as_view()),
+
+    # re_path(r'^application/uploadChallenge/(?P<filename>[^/]+)$', applications.UploadApplicationView.as_view())
```
Comment (Marcodaum) [inline backend/codingchallenge/api/urls.py:27]: remove comment

File: backend/codingchallenge/api/views/applications.py [hunk 2]
```diff
@@ -223,30 +235,114 @@ def delete(self, request, *args, **kwargs):
...
+                applicationId
+        """
+        try:
+            application = Application.objects.get(applicationId=self.kwargs["applicationId"])
+
+            if not application:
+                raise TypeError
+
+        except(KeyError, TypeError):
+            return Response(status=status.HTTP_404_NOT_FOUND)
```
Comment (Marcodaum) [inline backend/codingchallenge/api/views/applications.py:262]: No text gets returned (e.g. "Application ID not found!")

File: backend/codingchallenge/api/views/applications.py [hunk 3]
```diff
@@ -223,30 +235,114 @@ def delete(self, request, *args, **kwargs):
...
+
+        if application.githubRepo:
+            repoName = application.githubRepo
+        else:
+            return Response(
+                jsonMessages.errorJsonResponse("Can not find repo"),
+                status=status.HTTP_400_BAD_REQUEST)
```
Comment (Marcodaum) [inline backend/codingchallenge/api/views/applications.py:269]: no text gets returned

File: backend/codingchallenge/api/views/applications.py [hunk 4]
```diff
@@ -223,30 +235,114 @@ def delete(self, request, *args, **kwargs):
...
+                    self.gApi.pushFile(repoName, path, file_obj.read(path))
+
+            self.gApi.addLinter(repoName)
+
+            user.application.submission = time.time()
+            user.application.status = Application.Status.IN_REVIEW
+            user.application.githubRepo = repoName
+            user.application.save()
+
+            return Response({"success": "true"}, status=status.HTTP_200_OK)
```
Comment (Marcodaum) [inline backend/codingchallenge/api/views/applications.py:312]: use jsonMessages. ... instead of new success object

**Review by Marcodaum — CHANGES_REQUESTED** [review]
changes requested in comments

File: backend/codingchallenge/api/views/applications.py (hunk 4 above)
Comment (lorenzschreyer) [reply backend/codingchallenge/api/views/applications.py]: done

File: backend/codingchallenge/api/views/applications.py (hunk 3 above)
Comment (lorenzschreyer) [reply backend/codingchallenge/api/views/applications.py:269]: done

File: backend/codingchallenge/api/urls.py (hunk 1 above)
Comment (lorenzschreyer) [reply backend/codingchallenge/api/urls.py]: done

File: backend/codingchallenge/api/urls.py [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # from django.conf.urls import url
-from django.urls import path, include
+from django.urls import path, include, re_path
```
Comment (lorenzschreyer) [inline backend/codingchallenge/api/urls.py:2]: remove unused dependency

Comment (lorenzschreyer) [reply backend/codingchallenge/api/urls.py]: done

**Review by Marcodaum — APPROVED** [review]
passt

**Pull request merged**